NMS-13584: Notification Commands Documentation (#4596)
OpenNMS/opennms: docs/modules/operation/pages/notifications/commands.adoc

| PARAM_DESTINATION
| -d
| From notification definition

| PARAM_EMAIL
| -email
| From user profile

| PARAM_HOME_PHONE
| -hphone
| From user profile

| PARAM_INTERFACE
| -interface
| From original event

| PARAM_MICROBLOG_USERNAME
| -ublog

| PARAM_MOBILE_PHONE
| -mphone
| From user profile

| PARAM_NODE
| -nodeid
| From original event

| PARAM_NUM_MSG
| -nm
| From notification definition

| PARAM_NUM_PAGER_PIN
| -np
| From user profile

| PARAM_PAGER_EMAIL
| -pemail
| From user profile

| PARAM_RESPONSE
| -r
| From notification definition

| PARAM_SERVICE
| -service
| From original event

| PARAM_SUBJECT
| -subject
| From notification definition

| PARAM_TEXT_MSG
| -tm
| From notification definition

| PARAM_TEXT_PAGER_PIN
| -tp
| From user profile

| PARAM_TUI_PIN
| -tuipin

| PARAM_TYPE
| -t
| Binary y/n

| PARAM_TUI_PIN
| -tuipin
| From user profile

| PARAM_WORK_PHONE
| -wphone
| From user profile

| PARAM_XMPP_ADDRESS
| -xmpp
| From user profile

|===

=== Extended dynamic values

You can pass arbitrary parameters from the notification to the notificationCommand via the <parameter> tag, which can then be used as required.

== Examples

=== SnmpTrapNotificationStrategy

> This configuration sends an OpenNMS trap to an external system with the `nodelabel` in the varbind.

Reviewer: Let's use OpenNMS.

        <name>snmpTrap</name>
        <execute>org.opennms.netmgt.notifd.SnmpTrapNotificationStrategy</execute>
        <comment>Class for sending notifications as SNMP Traps</comment>
        <argument streamed="false">
                <switch>trapVersion</switch>
        </argument>
        <argument streamed="false">
                <switch>trapTransport</switch>
        </argument>
        <argument streamed="false">
                <switch>trapHost</switch>
        </argument>
        <argument streamed="false">
                <switch>trapPort</switch>
        </argument>
        <argument streamed="false">
                <switch>trapCommunity</switch>
        </argument>
        <argument streamed="false">
                <switch>trapEnterprise</switch>
        </argument>
        <argument streamed="false">
                <switch>trapGeneric</switch>
        </argument>
        <argument streamed="false">
                <switch>trapSpecific</switch>
        </argument>
        <argument streamed="false">
                <switch>trapVarbind</switch>
        </argument>
    </command>
----

You can define any or none of these parameters and switches.

> However, if you define it as a parameter in the notification, you must define it as a switch in the notification command.

Reviewer 2: ```suggestion
However, if you define a parameter in the notification, you must also define a corresponding switch in the notification command.
```

The defaults are as follows:

[source, properties]
----
trapVersion="v1" ("v1 or "v2c")
trapTransport="UDP" (only UDP for now)
trapHost="127.0.0.1" (any valid hostname or ip address)
trapPort="162" (any valid IP port)
trapCommunity="public" (any valid community string)
trapEnterprise=".1.3.6.1.4.1.5813" (any valid OID)
trapGeneric="6" (any valid generic ID)
trapSpecific="1" (any valid trapSpecific ID)
trapVarbind="OpenNMS Trap Notification" (any string or one or more valid %<event field>% such as "%eventUEI%" in the string)
----

> The one allowed trap varbind will be sent with the object ID `.1.3.6.1.4.1.5813.20.1` and the object type `DisplayString`.

Reviewer 2: ```suggestion
The one allowed trap varbind will be sent with the object ID `.1.3.6.1.4.1.5813.20.1` and the object type `DisplayString`.

```
